app/tests/app/page.test.tsx

```diff
@@ -1,43 +1,52 @@
 import { axe } from "jest-axe";
-import { GetServerSidePropsContext } from "next";
-import Index, { getServerSideProps } from "src/pages/index";
+import Controller from "src/app/[locale]/page";
+import { View } from "src/app/[locale]/view";
 import { LocalFeatureFlagManager } from "src/services/feature-flags/LocalFeatureFlagManager";
-import { render, screen } from "tests/react-utils";
+import { cleanup, render, screen } from "tests/react-utils";
 
-describe("Index", () => {
+describe("Index - Controller", () => {
+  it("retrieves feature flags", async () => {
+    const featureFlagSpy = jest
+      .spyOn(LocalFeatureFlagManager.prototype, "isFeatureEnabled")
+      .mockResolvedValue(true);
+
+    const result = await Controller();
+    render(result);
+
+    expect(featureFlagSpy).toHaveBeenCalledWith("foo", "anonymous");
+    expect(screen.getByText(/Flag is enabled/)).toBeInTheDocument();
+  });
+});
+
+describe("Index - View", () => {
   // Demonstration of rendering translated text, and asserting the presence of a dynamic value.
   // You can delete this test for your own project.
   it("renders link to Next.js docs", () => {
-    render(<Index isFooEnabled={true} />);
+    render(<View isFooEnabled={false} />);
 
     const link = screen.getByRole("link", { name: /next\.js/i });
 
     expect(link).toBeInTheDocument();
     expect(link).toHaveAttribute("href", "https://nextjs.org/docs");
+    expect(screen.getByText(/\$1,234/)).toBeInTheDocument();
   });
 
   it("passes accessibility scan", async () => {
-    const { container } = render(<Index isFooEnabled={true} />);
+    const { container } = render(<View isFooEnabled={false} />);
     const results = await axe(container);
 
     expect(results).toHaveNoViolations();
   });
 
   it("conditionally displays content based on feature flag values", () => {
-    const { container } = render(<Index isFooEnabled={true} />);
-    expect(container).toHaveTextContent("Flag is enabled");
-  });
+    const enabledFlagTextMatcher = /Flag is enabled/;
 
-  it("retrieves feature flags", async () => {
-    const featureName = "foo";
-    const userId = "anonymous";
-    const featureFlagSpy = jest
-      .spyOn(LocalFeatureFlagManager.prototype, "isFeatureEnabled")
-      .mockResolvedValue(true);
-    await getServerSideProps({
-      req: { cookies: {} },
-      res: {},
-    } as unknown as GetServerSidePropsContext);
-    expect(featureFlagSpy).toHaveBeenCalledWith(featureName, userId);
+    render(<View isFooEnabled />);
+    expect(screen.getByText(enabledFlagTextMatcher)).toBeInTheDocument();
+
+    cleanup();
```

Comment:
I was wondering how to do this before and couldn't find the cleanup function! Nice to see this.